geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -178,19 +186,23 @@ public LocatorClusterManagementService(ConfigurationPersistenceService persisten
     }
 
     // persist configuration in cache config
-    persistenceService.updateCacheConfig(groupName, cacheConfigForGroup -> {
-      try {
-        configurationManager.add(config, cacheConfigForGroup);
-        result.setStatus(StatusCode.OK,
-            "Successfully updated configuration for " + groupName + ".");
-      } catch (Exception e) {
-        String message = "Failed to update cluster configuration for " + groupName + ".";
-        logger.error(message, e);
-        result.setStatus(StatusCode.FAIL_TO_PERSIST, message);
-        return null;
-      }
-      return cacheConfigForGroup;
-    });
+    if (configurationManager instanceof CacheConfigurationManager) {
+      persistenceService.updateCacheConfig(groupName, cacheConfigForGroup -> {
+        try {
+          ((CacheConfigurationManager) configurationManager).add(config, cacheConfigForGroup);
+          result.setStatus(StatusCode.OK,
+              "Successfully updated configuration for " + groupName + ".");
+        } catch (Exception e) {
+          String message = "Failed to update cluster configuration for " + groupName + ".";
+          logger.error(message, e);
+          result.setStatus(StatusCode.FAIL_TO_PERSIST, message);
+          return null;
+        }
+        return cacheConfigForGroup;
+      });
+    } else if (configurationManager instanceof DeploymentManager) {
+      // invoke persistenceService.addJarsToThisLocator(....);
```

Comment:
should this be commented out? If not then can this code be moved to DeploymentManager to prevent instanceof check?